opal/class/opal_object.h

```diff
@@ -468,7 +468,7 @@ static inline void opal_obj_run_destructors(opal_object_t *object)
     assert(NULL != object->obj_class);
 
     cls_destruct = object->obj_class->cls_destruct_array;
-    while (NULL != *cls_destruct) {
+    while (NULL != cls_destruct) {
```

Comment:
How it is possible to have a zero-level class hierarchy for any object other than opal_class_t ? It looks as if you are trying to free a class that was not yet initialized.